### PR TITLE
feat: add system resource monitoring with TUI and install integration

### LINKED ISSUE #51: Real-Time System Health Monitor
## Description
Monitor system resources (CPU, RAM, disk, network) in real-time during installations.

## Requirements
- Real-time CPU/memory/disk monitoring
- Installation impact tracking
- Resource usage alerts
- Historical resource usage
- Performance recommendations
- Export monitoring data

## Acceptance Criteria
- [ ] Monitors CPU, RAM, disk, network usage
- [ ] Shows real-time metrics during installs
- [ ] Alerts on resource constraints
- [ ] Stores historical data
- [ ] Suggests performance improvements
- [ ] Export metrics to CSV/JSON
- [ ] Unit tests included (>80% coverage)
- [ ] Documentation with monitoring examples

## Example Usage
```bash
$ cortex monitor
System Health:
  CPU:     45% (4 cores)
  RAM:     8.2/16 GB (51%)
  Disk:    120/500 GB (24%)
  Network: 2.5 MB/s ↓  0.8 MB/s ↑

$ cortex install cuda --monitor
Installing CUDA...
  CPU: ████████░░ 80% (compilation)
  RAM: ██████████ 12.5/16 GB
  Disk: Writing... 2.1 GB/3.5 GB
  
✓  Installation complete
📊 Peak usage: CPU 95%, RAM 13.2 GB
```

**Skills:** Python, psutil, system monitoring
**Bounty:** $25 upon merge
**Priority:** High

### PR DESCRIPTION
**Pull request by Vaibhav701161**
## Related Issue

Closes #51

---

## Summary

This PR introduces **system resource monitoring** to Cortex, providing visibility into CPU, RAM, Disk, and Network usage both **standalone** and **during installations**.

### What’s included

* **Standalone monitoring**

  * `cortex monitor` launches a real-time TUI dashboard
  * Optional duration and export support (JSON / CSV)
* **Install-time monitoring**

  * `cortex install <package> --monitor`
  * Captures and reports **peak resource usage** during installation
* **Persistent storage**

  * Monitoring sessions and samples stored in SQLite (backward-compatible schema)
* **Advisory analysis**

  * Simple rule-based recommendations for resource pressure
* **Graceful fallbacks**

  * Safe behavior when `psutil` or `rich` is unavailable
* **Memory-safe design**

  * Bounded sample storage to prevent unbounded growth

---

## Implementation Overview

* New `cortex/monitor/` module:

  * `sampler.py` – thread-safe resource sampling via `psutil`
  * `monitor_ui.py` – real-time TUI using `rich.Live`
  * `storage.py` – SQLite-backed persistence
  * `exporter.py` – JSON / CSV export
  * `analyzer.py` – advisory recommendations
* CLI integration:

  * New `monitor` command
  * `--monitor` flag wired through `install` → `InstallationCoordinator`
* Backward-compatible extension of `InstallationResult` to include peak metrics

* `Implementation`

image

* `Export functionality`

image

* `Monitoring Installation`

image

* `Stress Check`

https://github.com/user-attachments/assets/d0f40c52-8339-477c-bce8-1bf8ccebff28

---

## Testing

* Comprehensive unit tests for:

  * Sampler internals (threading, callbacks, error paths)
  * Analyzer edge cases and trends
  * Storage lifecycle (create, save, list, finalize, delete)
  * CLI argument wiring
  * Monitor UI (init, rendering helpers, fallback mode)
* Manual verification:

  * Live TUI monitoring
  * Install-time monitoring with peak summary
  * Exported JSON / CSV validation
* Coverage target met (>80% for `cortex/monitor/`)

https://github.com/user-attachments/assets/fc0ab79b-e978-400a-a9c9-f17d34c8756a

---

## Non-Goals (Intentional)

* Per-process attribution
* Daemon-side monitoring
* GPU monitoring
* Automatic install cancellation

---

## AI Disclosure

* [ ] No AI used
* [x] AI/IDE/Agents used (please describe below)

Used AI assistance (Google Antigravity) for:

* Reviewing architectural trade-offs
* Identifying edge cases and failure modes
* Generating and refining unit test scenarios
* Improving documentation clarity

All code, design decisions, and final implementations were reviewed, validated, and iterated manually.

---

## Checklist

* [x] PR title follows format: `feat(monitor): add system resource monitoring`
* [x] Tests pass (`pytest tests/`)
* [ ] MVP label added if closing MVP issue
* [x] Update "Cortex -h" (CLI help updated)

## Summary by CodeRabbit

* **New Features**
  * Real-time interactive monitor (standalone command with duration/interval/export), install-time --monitor flag that prints peak CPU/RAM after installs, live UI, exports (JSON/CSV), persistent local session storage, and automated analysis with recommendations/warnings.

* **Tests**
  * Extensive unit and integration tests covering sampler, UI, exporter, analyzer, storage, and CLI integration.

* **Chores**
  * Added psutil as a runtime dependency.

* **Documentation**
  * New monitoring guide detailing capabilities, CLI usage, exports, and fallback behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by Anshgrover23 — CHANGES_REQUESTED** [review]
@Vaibhav701161 Thanks for contributing to Cortex Linux.
Please open a separate PR for CLA verification, following the pattern used in #401.

Also, kindly address all bot comments.

File: pyproject.toml [hunk 1]
```diff
@@ -56,13 +56,15 @@ dependencies = [
     "cryptography>=42.0.0",
     # Terminal UI and formatting
     "rich>=13.0.0",
+    # System resource monitoring (required for cortex monitor)
+    "psutil>=5.9.0",
```
[the remark this answers is not in the document]
Comment (Vaibhav701161) [reply pyproject.toml]: fixed

File: cortex/monitor/monitor_ui.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+        result.append(label_text, style="bold")
+        result.append(bar, style=color)
+        result.append(f" {percent:5.1f}%", style=color)
+
+        if cores:
+            result.append(f" ({cores} cores)", style="dim")
+        elif suffix:
+            result.append(f" ({suffix})", style="dim")
+
+        return result
```
[the remark this answers is not in the document]
Comment (Vaibhav701161) [reply cortex/monitor/monitor_ui.py]: fixed

File: cortex/monitor/monitor_ui.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+SPDX-License-Identifier: BUSL-1.1
+"""
+
+import logging
+import sys
+import time
+from datetime import datetime
+
+from cortex.monitor.sampler import AlertThresholds, PeakUsage, ResourceSample, ResourceSampler
+
```
[the remark this answers is not in the document]
Comment (Vaibhav701161) [reply cortex/monitor/monitor_ui.py]: fixed

File: cortex/monitor/monitor_ui.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+            color = "green"
+
+        # Build bar
+        filled = int((percent / 100) * BAR_WIDTH)
+        bar = "█" * filled + "░" * (BAR_WIDTH - filled)
+
+        # Format label
+        label_text = f"{label:>6}: "
+        if cores:
+            label_text = f"{label:>6}: "
```
[the remark this answers is not in the document]
Comment (Vaibhav701161) [reply cortex/monitor/monitor_ui.py]: fixed
